plugins/ui-update/components/InstalledVersion.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from "react";
+import { Trans } from "@lingui/macro";
+import semver from "semver";
+
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import { UIMetadata } from "#SRC/js/data/ui-update/UIMetadata";
+import {
+  UIAction,
+  UIActions,
+  UIActionType
+} from "#PLUGINS/ui-update/types/UIAction";
+
+interface InstalledVersionProps {
+  uiMetaData: UIMetadata | null;
+  uiAction?: UIAction;
+  onRollbackClick?: () => void;
+}
+function refreshPage() {
+  location.reload();
+}
+
+function renderRollbackAction(uiAction: UIAction): React.ReactNode {
+  let defaultContent = <Trans>Rollback</Trans>;
+  if (uiAction.type !== UIActionType.Reset) {
+    return defaultContent;
+  }
+  switch (uiAction.action) {
+    case UIActions.Started:
+      return <Trans>Rolling back...</Trans>;
+    case UIActions.Error:
+      return <Trans>Rollback Failed!</Trans>;
+    default:
+      return defaultContent;
+  }
+}
+
+function rollbackActionDisabled(uiAction: UIAction): boolean {
+  if (uiAction.type === UIActionType.None) {
+    return false;
+  }
+  if (uiAction.type === UIActionType.Reset) {
+    return (
+      uiAction.action === UIActions.Started ||
+      uiAction.action === UIActions.Error
+    );
+  }
+  return uiAction.action === UIActions.Started;
+}
+
+function renderRollbackContent(props: InstalledVersionProps) {
+  if (!props.uiAction) {
+    return null;
+  }
+  return (
+    <button
+      id="uiDetailsRollback"
+      className="button button-primary-link"
+      onClick={props.onRollbackClick}
+      disabled={rollbackActionDisabled(props.uiAction)}
+    >
+      {renderRollbackAction(props.uiAction)}
+    </button>
+  );
+}
+
+const InstalledVersion = (props: InstalledVersionProps) => {
+  const { uiMetaData } = props;
+  if (!uiMetaData || !uiMetaData.clientBuild) {
+    return null;
+  }
+  const { clientBuild, serverBuild } = uiMetaData;
+
+  let refreshForUpdateContent = null;
+
+  let coercedClientBuild = semver.coerce(clientBuild);
```

Comment:
This and all following variables could be const. You could also extract the button render to a separate render function which could help you get rid of the variable `refreshForUpdateContent`